subprojects/core/src/integTest/groovy/org/gradle/api/file/SettingsLayoutIntegrationTest.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.file
+
+import org.gradle.api.internal.file.DefaultSettingsLayout
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class SettingsLayoutIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
🤔  This is less important, but I think we should also add a test that uses `--settings-file` flag to specify the settings file location, and demonstrates the behavior of the settings layout in this case. The flag is deprecated, but the test should be there until we actually remove it.

Reply:
Done!